### PR TITLE
Aichat Sagemaker Helper and Model Processors refactor

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR is a follow-up to https://github.com/code-dot-org/code-dot-org/pull/59649 which updated input/output of Aichat fine-tuned models and https://github.com/code-dot-org/code-dot-org/pull/59760 which added tests.

This PR includes a small refactor in `aichat_sagemaker_helper` addressing @sanchitmalhotra126 's comment https://github.com/code-dot-org/code-dot-org/pull/59760#discussion_r1678099671.
It also creates a subdirectory in `helpers` titled `ai_model_processors` to help keep model processors in one place as it is likely there will be more fine-tuned models added in the future.

There is no change in behavior from the user-end. This is just an eng refactor.

## Testing story
Tested locally in aichat levels to make sure that behavior did not change after refactor.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/aichat_controller_test.rb [hunk 1]
```diff
@@ -97,11 +97,6 @@ class AichatControllerTest < ActionController::TestCase
       ]
     )
 
-    # Note that second expected argument filters out the previous profane message
-    # in what we send to Sagemaker.
-    AichatSagemakerHelper.expects(:format_inputs_for_sagemaker_request).
```
Comment (bencodeorg) [inline dashboard/test/controllers/aichat_controller_test.rb:102]: Is this ~~covered~~ test case covered elsewhere now?

Comment (fisher-alice) [reply dashboard/test/controllers/aichat_controller_test.rb:102]: Good call! I'll replace with statement testing that `get_sagemaker_assistant_response` was called with profane messages filtered out. Thanks!

File: dashboard/app/controllers/aichat_controller.rb [hunk 2]
```diff
@@ -41,20 +41,12 @@ def chat_completion
       }
     end
 
-    messages_for_model = params.to_unsafe_h[:storedMessages].filter do |message|
```
Comment (fisher-alice) [inline dashboard/app/controllers/aichat_controller.rb:44]: @bencodeorg - seems like unit tests are passing without `to_unsafe_h` so removed. Lmk what you think. Thanks!

File: dashboard/app/controllers/aichat_controller.rb [hunk 3]
```diff
@@ -41,20 +41,12 @@ def chat_completion
...
-    # Use to_unsafe_h here to allow testing this function.
-    # Safe params are primarily targeted at preventing "mass assignment vulnerability"
-    # which isn't relevant here.
-    inputs = AichatSagemakerHelper.format_inputs_for_sagemaker_request(
-      params.to_unsafe_h[:aichatModelCustomizations],
-      messages_for_model,
-      params.to_unsafe_h[:newMessage]
-    )
-    latest_assistant_response_from_sagemaker = AichatSagemakerHelper.get_sagemaker_assistant_response(inputs, params[:aichatModelCustomizations][:selectedModelId])
+    latest_assistant_response_from_sagemaker = AichatSagemakerHelper.get_sagemaker_assistant_response(params[:aichatModelCustomizations], messages_for_model, params[:newMessage])
```
Comment (sanchitmalhotra126) [inline dashboard/app/controllers/aichat_controller.rb:49]: 🎉

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM! Nice refactoring!

File: dashboard/app/controllers/aichat_controller.rb (hunk 2 above)
Comment (bencodeorg) [reply dashboard/app/controllers/aichat_controller.rb:44]: huh, [went back](https://github.com/code-dot-org/code-dot-org/commit/97dbf96e7043626ea3c704788fb00e91e52c1594) to where I added this and seems like the tests passed at this point, so maybe I totally misunderstood something at the time. Also did a quick verify that if I changed the expect condition to something else, it fails, so it seems like it's doing its job. Thanks for cleaning up!

**Review by bencodeorg — APPROVED** [review]
Nice!

**Pull request merged**